src/octoprint/util/yaml_util.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+import io
+
+
+def safe_load(file=None, path=None):
+    """
+    Safely and performantly loads yaml data from the given source. Either a
+    path or a file must be passed in.
+
+    :rtype: Union[dict[Hashable, Any], list, None]
+    :type path: str | None
+    :type file: typing.TextIO | None
+    """
+    if path is not None:
+        assert file is None
+        with io.open(path, "rt", encoding="utf-8") as f:
+            return safe_load(file=f)
+
+    if file is not None:
+        assert path is None
+
+    assert (path is not None) or (file is not None)
+
+    import yaml
+
+    try:
+        from yaml import CSafeLoader as SafeLoader
+    except ImportError:
+        from yaml import SafeLoader
+
+    return yaml.load(file, Loader=SafeLoader)
+
+
+def safe_dump(data, file=None, path=None, **kwargs):
```

Comment:
`save_to_file`

Reply:
Done, but it feels kinda weird because most clients aren't passing a file or a path and are getting a string back.

Reply:
Hm, I see what you mean. We actually have two uses here, dumping and saving to a file object or a path, and just dumping and returning the result. So maybe this should be two functions after all? 🤔

`dump` & `save_to_file` maybe?